### PR TITLE
action: release macos bianries as well

### PR DESCRIPTION
**Pull request by bergwolf**
Right now, only nydusctl and nydusd compile on macos. So just include
them in the release tarball. Other binaries can be included later.

Example run can be found here: https://github.com/bergwolf/image-service/runs/6420841588

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -80,46 +120,90 @@ jobs:
...
-  upload-artifacts:
+
+  prepare-tarball-linux:
     runs-on: ubuntu-latest
     strategy:
       matrix:
         arch: [amd64, arm64]
-    needs: [build-nydus-rs, build-contrib]
+        os: [linux]
+    needs: [nydus-linux, nydus-macos, contrib-linux]
```
Comment (imeoer) [inline .github/workflows/release.yml:137]: `nydus-macos` is unnecessary.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -80,46 +120,90 @@ jobs:
...
+        path: ${{ env.tarball }}
+
+  # use a seperate job for darwin because github action if: condition cannot handle && properly.
+  prepare-tarball-darwin:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [amd64]
+        os: [darwin]
+    needs: [nydus-linux, nydus-macos, contrib-linux]
```
Comment (imeoer) [inline .github/workflows/release.yml:164]: `nydus-linux` is unnecessary.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -80,46 +120,90 @@ jobs:
...
+    - name: push release
+      if: github.event_name == 'push'
       uses: softprops/action-gh-release@v1
       with:
         name: "Nydus Image Service ${{ env.tag }}"
         generate_release_notes: true
-        files:
-          ${{ env.tarball }}
+        files: |
+          ${{ env.tarball1 }}
```
Comment (imeoer) [inline .github/workflows/release.yml:207]: Can we use an `env.tarballs` value which is split by `\n` here to simplify this?

Comment (bergwolf) [reply .github/workflows/release.yml:207]: The newline will be eaten by $GITHUB_ENV file and we can only retrive a single line there.

Comment (bergwolf) [reply .github/workflows/release.yml]: fixed

**Review by imeoer — APPROVED** [review]
LGTM!

**Pull request merged**